presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadata.java

```diff
@@ -1786,6 +1798,137 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
         return Optional.of(new ConstraintApplicationResult<>(newHandle, partitionResult.getUnenforcedConstraint()));
     }
 
+    @Override
+    public Optional<ProjectionApplicationResult<ConnectorTableHandle>> applyProjection(ConnectorSession session, ConnectorTableHandle handle, List<ConnectorExpression> projections, Map<String, ColumnHandle> assignments)
+    {
+        if (projections.isEmpty()) {
+            return Optional.empty();
+        }
+
+        SchemaTableName tableName = ((HiveTableHandle) handle).getSchemaTableName();
+        Table table = metastore.getTable(new HiveIdentity(session), tableName.getSchemaName(), tableName.getTableName())
+                .orElseThrow(() -> new TableNotFoundException(tableName));
+
+        // Support ORC tables only
+        if (!extractHiveStorageFormat(table).equals(HiveStorageFormat.ORC)) {
```

Comment:
The format can vary partition by partition, so it needs to be supported for all readers.